src/sxg_buffer.c

```diff
@@ -111,6 +111,27 @@ bool sxg_write_byte(uint8_t byte, sxg_buffer_t* target) {
   return true;
 }
 
+bool sxg_write_utf8_cbor_header(size_t length, sxg_buffer_t* target) {
+  if (length <= 0x17) {
+    return sxg_write_byte(0x60 + length, target);
+  } else if (length <= 0xff) {
+    return sxg_write_byte(0x78, target) && sxg_write_int(length, 1, target);
+  } else if (length <= 0xffff) {
+    return sxg_write_byte(0x79, target) && sxg_write_int(length, 2, target);
+  } else if (length <= 0xffffffff) {
```

Comment:
I think this promotes correctly but it might be better to have a `0xffffffffULL` constant for clarity.